File: writer-reader-backend/app/Models/Comment.php
<?php

namespace App\Models;

use Database\Factories\WorkFactory;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\MorphMany;
use Illuminate\Database\Eloquent\Relations\MorphTo;

/**
 * @class Comment
 * @package App\Models
 * @property string id
 * @property string content
 * @property int user_id
 * @property string commentable_id
 * @property string commentable_type
 * @property string created_at
 * @property string updated_at
 * @property User user
 */
class Comment extends Model
{
    /** @use HasFactory<WorkFactory> */
    use HasFactory;
    use HasUuids;

    protected $fillable = [
        'content',
        'user_id',
        'commentable_id',
        'commentable_type',
        'created_at',
        'updated_at',
    ];

    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function commentable(): MorphTo
    {
        return $this->morphTo();
    }

    public function likes(): MorphMany
    {
        return $this->morphMany(Like::class, 'likeable');
    }
}

File: writer-reader-android-client/app/src/main/java/hu/bme/aut/android/writer_reader_client/data/model/get/Comment.kt
package hu.bme.aut.android.writer_reader_client.data.model.get

import com.squareup.moshi.Json


data class Comment(
    val id: String = "",
    val content: String = "",
    @Json(name = "user_id") val userId: String = "",
    @Json(name = "user_name") val userName: String = "",
    @Json(name = "commentable_type") val commentableType: String = "",
    @Json(name = "commentable_id") val commentableId: String = "",
    @Json(name  = "created_at") val createdAt: String = "",
    @Json(name = "updated_at") val updatedAt: String = "",
    val likes: Int = 0,
    @Json(name = "is_liked") val isLiked: Boolean = false,
)